### ISSUE TITLE
useSystemNetMailSettings=false still uses .config (partially) 

### ISSUE THREAD

**Issue by yang7749** [NLog/NLog#690]
MailTarget's useSystemNetMailSettings parameter doesn't seem to work.

NLog.config:
```
<target xsi:type="Mail" name="Email" from="test@test.com" to="test@test.com" useSystemNetMailSettings="false"
            smtpServer="localhost" smtpPort="25" timeout="30000"
            subject="test" 
            body="test" />
```

web.config:
```
<system.net>
        <mailSettings>
            <smtp deliveryMethod="SpecifiedPickupDirectory">
                <specifiedPickupDirectory pickupDirectoryLocation="C:\TEMP\email_out" />
            </smtp>
        </mailSettings>
    </system.net>
```

The logger still add files to "C:\TEMP\email_out" whenever it tries to send email

**Comment by 304NotModified** [comment]
Please post the internal log file. Thanks!

https://github.com/NLog/NLog/blob/master/CONTRIBUTING.md#bug-reports

**Comment by yang7749** [comment]
There aren't any message in the internal log related to it. Here is all I see in the internal log after my code logs an error that should have send the email:

```
2015-05-04 16:22:01.3895 Info Adding target Mail Target[Email]
2015-05-04 16:22:01.4085 Info Configured from an XML element in C:\SVN\TestWebsite\bin\NLog.config...
2015-05-04 16:22:01.4085 Info Watching path 'C:\SVN\TestWebsite\bin' filter 'NLog.config' for changes.
2015-05-04 16:22:01.4085 Info Watching path 'c:\' filter 'NLog.config' for changes.
2015-05-04 16:22:01.4245 Info Found 12 configuration items
```

**Comment by 304NotModified** [comment]
Also not in `debug` level?

**Comment by 304NotModified** [comment]
Or even `trace`.  It would be odd if it doesn't work because we use the default .net class for this. https://msdn.microsoft.com/en-us/library/k1c4h6e2(v=vs.110).aspx

**Comment by yang7749** [comment]
Here is my internal log with trace:
```
2015-05-06 10:42:48.3925 Trace FindReachableObject<NLog.Internal.IRenderable>:
2015-05-06 10:42:48.3985 Trace Scanning LongDateLayoutRenderer 'Layout Renderer: ${longdate}'
2015-05-06 10:42:48.3985 Debug Setting 'UppercaseLayoutRendererWrapper.uppercase' to 'true'
2015-05-06 10:42:48.4135 Trace Wrapping LevelLayoutRenderer with UppercaseLayoutRendererWrapper
2015-05-06 10:42:48.4135 Trace FindReachableObject<NLog.Internal.IRenderable>:
2015-05-06 10:42:48.4135 Trace Scanning LevelLayoutRenderer 'Layout Renderer: ${level}'
2015-05-06 10:42:48.4135 Trace FindReachableObject<NLog.Internal.IRenderable>:
2015-05-06 10:42:48.4135 Trace Scanning UppercaseLayoutRendererWrapper 'Layout Renderer: ${uppercase}'
2015-05-06 10:42:48.4135 Trace  Scanning SimpleLayout ''''
2015-05-06 10:42:48.4135 Trace   Scanning LevelLayoutRenderer 'Layout Renderer: ${level}'
2015-05-06 10:42:48.4135 Trace FindReachableObject<NLog.Internal.IRenderable>:
2015-05-06 10:42:48.4135 Trace Scanning LoggerNameLayoutRenderer 'Layout Renderer: ${logger}'
2015-05-06 10:42:48.4135 Trace FindReachableObject<NLog.Internal.IRenderable>:
2015-05-06 10:42:48.4296 Trace Scanning MessageLayoutRenderer 'Layout Renderer: ${message}'
2015-05-06 10:42:48.4296 Trace FindReachableObject<NLog.Internal.IRenderable>:
2015-05-06 10:42:48.4296 Trace Scanning MessageLayoutRenderer 'Layout Renderer: ${message}'
2015-05-06 10:42:48.4296 Trace FindReachableObject<NLog.Internal.IRenderable>:
2015-05-06 10:42:48.4296 Trace Scanning NewLineLayoutRenderer 'Layout Renderer: ${newline}'
2015-05-06 10:42:48.4296 Trace FindReachableObject<NLog.Internal.IRenderable>:
2015-05-06 10:42:48.4296 Trace Scanning MachineNameLayoutRenderer 'Layout Renderer: ${machinename}'
2015-05-06 10:42:48.4486 Debug Setting 'MailTarget.name' to 'Email'
2015-05-06 10:42:48.4486 Debug Setting 'MailTarget.from' to 'test@test.com'
2015-05-06 10:42:48.4486 Debug Setting 'MailTarget.to' to 'test@test.com'
2015-05-06 10:42:48.4486 Debug Setting 'MailTarget.smtpServer' to 'localhost'
2015-05-06 10:42:48.4486 Debug Setting 'MailTarget.smtpPort' to '25'
2015-05-06 10:42:48.4486 Debug Setting 'MailTarget.timeout' to '30000'
2015-05-06 10:42:48.4606 Debug Setting 'MailTarget.useSystemNetMailSettings' to 'false'
2015-05-06 10:42:48.4606 Debug Setting 'MailTarget.subject' to 'test'
2015-05-06 10:42:48.4606 Debug Setting 'MailTarget.body' to 'test'
2015-05-06 10:42:48.4606 Info Adding target Mail Target[Email]
2015-05-06 10:42:48.4606 Debug Registering target Email: NLog.Targets.MailTarget
2015-05-06 10:42:48.4606 Trace ParseRulesElement
2015-05-06 10:42:48.4826 Info Configured from an XML element in C:\TestWebsite\bin\NLog.config...
2015-05-06 10:42:48.4826 Debug --- NLog configuration dump. ---
2015-05-06 10:42:48.4826 Debug Targets:
2015-05-06 10:42:48.4926 Info Mail Target[Email]
2015-05-06 10:42:48.4926 Debug Rules:
2015-05-06 10:42:48.4926 Info logNamePattern: (:All) levels: [ ] appendTo: [ Email ]
2015-05-06 10:42:48.4926 Debug --- End of NLog configuration dump ---
2015-05-06 10:42:48.4926 Info Watching path 'C:\TestWebsite\bin' filter 'NLog.config' for changes.
2015-05-06 10:42:48.5076 Info Watching path 'C:\' filter 'NLog.config' for changes.
2015-05-06 10:42:48.5076 Trace FindReachableObject<System.Object>:
2015-05-06 10:42:48.5076 Trace Scanning LoggingRule 'logNamePattern: (:All) levels: [ ] appendTo: [ Email ]'
2015-05-06 10:42:48.5076 Trace  Scanning MailTarget 'Mail Target[Email]'
2015-05-06 10:42:48.5076 Trace   Scanning SimpleLayout ''test@test.com''
2015-05-06 10:42:48.5226 Trace    Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.5226 Trace   Scanning SimpleLayout ''test@test.com''
2015-05-06 10:42:48.5226 Trace    Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.5226 Trace   Scanning SimpleLayout ''test''
2015-05-06 10:42:48.5226 Trace    Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.5226 Trace   Scanning SimpleLayout ''test''
2015-05-06 10:42:48.5226 Trace    Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.5396 Trace   Scanning SimpleLayout ''localhost''
2015-05-06 10:42:48.5396 Trace    Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.5396 Info Found 12 configuration items
2015-05-06 10:42:48.5396 Trace Initializing Layout Renderer: ${literal}
2015-05-06 10:42:48.5396 Trace Initializing 'localhost'
2015-05-06 10:42:48.5576 Trace FindReachableObject<System.Object>:
2015-05-06 10:42:48.5576 Trace Scanning SimpleLayout ''localhost''
2015-05-06 10:42:48.5576 Trace  Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.5576 Trace Initializing Layout Renderer: ${literal}
2015-05-06 10:42:48.5576 Trace Initializing 'test'
2015-05-06 10:42:48.5696 Trace FindReachableObject<System.Object>:
2015-05-06 10:42:48.5696 Trace Scanning SimpleLayout ''test''
2015-05-06 10:42:48.5696 Trace  Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.5696 Trace Initializing Layout Renderer: ${literal}
2015-05-06 10:42:48.5696 Trace Initializing 'test'
2015-05-06 10:42:48.5696 Trace FindReachableObject<System.Object>:
2015-05-06 10:42:48.5696 Trace Scanning SimpleLayout ''test''
2015-05-06 10:42:48.5856 Trace  Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.5856 Trace Initializing Layout Renderer: ${literal}
2015-05-06 10:42:48.5856 Trace Initializing 'test@test.com'
2015-05-06 10:42:48.5856 Trace FindReachableObject<System.Object>:
2015-05-06 10:42:48.5856 Trace Scanning SimpleLayout ''test@test.com''
2015-05-06 10:42:48.5856 Trace  Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.6006 Trace Initializing Layout Renderer: ${literal}
2015-05-06 10:42:48.6006 Trace Initializing 'test@test.com'
2015-05-06 10:42:48.6006 Trace FindReachableObject<System.Object>:
2015-05-06 10:42:48.6006 Trace Scanning SimpleLayout ''test@test.com''
2015-05-06 10:42:48.6006 Trace  Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.6006 Trace Initializing Mail Target[Email]
2015-05-06 10:42:48.6166 Trace FindReachableObject<NLog.Layouts.Layout>:
2015-05-06 10:42:48.6166 Trace Scanning MailTarget 'Mail Target[Email]'
2015-05-06 10:42:48.6166 Trace  Scanning SimpleLayout ''test@test.com''
2015-05-06 10:42:48.6166 Trace   Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.6166 Trace  Scanning SimpleLayout ''test@test.com''
2015-05-06 10:42:48.6166 Trace   Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.6326 Trace  Scanning SimpleLayout ''localhost''
2015-05-06 10:42:48.6326 Trace   Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.6326 Trace  Scanning SimpleLayout ''test''
2015-05-06 10:42:48.6326 Trace   Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.6326 Trace  Scanning SimpleLayout ''test''
2015-05-06 10:42:48.6326 Trace   Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.6476 Trace Mail Target[Email] has 5 layouts
2015-05-06 10:42:48.6676 Trace FindReachableObject<NLog.Internal.IUsesStackTrace>:
2015-05-06 10:42:48.6676 Trace Scanning TargetWithFilterChain 'NLog.Internal.TargetWithFilterChain'
2015-05-06 10:42:48.6676 Trace  Scanning MailTarget 'Mail Target[Email]'
2015-05-06 10:42:48.6676 Trace   Scanning SimpleLayout ''test@test.com''
2015-05-06 10:42:48.6796 Trace    Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.6796 Trace   Scanning SimpleLayout ''test@test.com''
2015-05-06 10:42:48.6796 Trace    Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.6796 Trace   Scanning SimpleLayout ''localhost''
2015-05-06 10:42:48.6796 Trace    Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.6796 Trace   Scanning SimpleLayout ''test''
2015-05-06 10:42:48.6956 Trace    Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:48.6956 Trace   Scanning SimpleLayout ''test''
2015-05-06 10:42:48.6956 Trace    Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2015-05-06 10:42:50.3957 Debug Targets for _Default by level:
2015-05-06 10:42:50.3957 Debug Trace =>
2015-05-06 10:42:50.4127 Debug Debug =>
2015-05-06 10:42:50.4127 Debug Info =>
2015-05-06 10:42:50.4127 Debug Warn =>
2015-05-06 10:42:50.4127 Debug Error => Email
2015-05-06 10:42:50.4258 Debug Fatal => Email
2015-05-06 10:45:11.1728 Debug Sending mail to test@test.com using localhost:25 (ssl=False)
2015-05-06 10:45:11.1768 Trace   Subject: 'test'
2015-05-06 10:45:11.1768 Trace   From: 'test@test.com'
2015-05-06 10:45:12.9610 Debug Sending mail to test@test.com using localhost:25 (ssl=False)
2015-05-06 10:45:12.9610 Trace   Subject: 'test'
2015-05-06 10:45:12.9740 Trace   From: 'test@test.com'
2015-05-06 10:45:31.3228 Debug Sending mail to test@test.com using localhost:25 (ssl=False)
2015-05-06 10:45:31.3228 Trace   Subject: 'test'
2015-05-06 10:45:31.3228 Trace   From: 'test@test.com'

```

**Comment by 304NotModified** [comment]
I see the issue. `useSystemNetMailSettings=true` is working, but `useSystemNetMailSettings=false` still use some settings of the config. This has to due to the default constructor [SmtpClient](https://msdn.microsoft.com/en-us/library/k1c4h6e2(v=vs.110).aspx) class. 

We should use another constructor of `SmtpClient`

**Comment by Xharze** [comment]
The pickup directory cannot be overridden by using any constructor. I think we have to add this configuration value to the target to be able to override this.

**Comment by 304NotModified** [comment]
That's also an option. 

But it seems that the other 2 [ctors of SMtpClient](https://msdn.microsoft.com/en-us/library/System.Net.Mail.SmtpClient.SmtpClient(v=vs.110).aspx), won't read the .config. That saves overwriting stuff - which isn't robust for future changes in `SmtpClient`

**Comment by Xharze** [comment]
@304NotModified I read it as the host and port won't be read from the config file, but other settings will. We'll have to try it, to which is correct.

**Comment by 304NotModified** [comment]
@Xharze your are right! So different ctor won't change a lot. 

Just checked the source at http://referencesource.microsoft.com/#System/net/System/Net/mail/SmtpClient.cs,ebaa2d61cfed8051

**Comment by dnlgmzddr** [comment]
I will take care of this issue.

**Comment by dnlgmzddr** [comment]
@304NotModified I was reviewing the issue, and we will need to add a new setting to specified the "PickupDirectoryLocation" override, but also this setting open a flow for the creation of the SMTP client.

With a pickup directory, we no longer require a server and the presence of both we raise a new exception. Should we use the same logic used by SmptClient which is:

if there's a server use it if not look for the pickup directory and if is not present throw an exception.

**Comment by 304NotModified** [comment]
Sounds good @dnlgmzddr

**Comment by dnlgmzddr** [comment]
@304NotModified Is there anything else to do in this issue?

**Comment by 304NotModified** [comment]
Nope, it has been released in 4.2.0, but we forgot the 'fixes' tag, see https://help.github.com/articles/closing-issues-via-commit-messages/

**Comment by 304NotModified** [comment]
Thanks for the reminder Daniel!
